Client/Frontend/Sync/BraveCore/Bookmarks/BookmarkManager.swift

```diff
@@ -184,6 +184,21 @@ class BookmarkManager {
         })
     }
     
+    public func fetchBookmarks(with query: String = "", completion: @escaping ([Bookmarkv2]) -> Void) {
+        guard let bookmarksAPI = bookmarksAPI else {
+            completion([])
+            return
+        }
+        
+        bookmarksAPI.search(withQuery: query, maxCount: 200, completion: { nodes in
+            var fetchedBookmarks: [Bookmarkv2] = []
+
+            fetchedBookmarks = nodes.compactMap({ return !$0.isFolder ? Bookmarkv2($0) : nil })
```

Comment:
The problem is it will need another run loop for that, this will work faster but for 200 entries that change in fetch speed might be ignorable. Up to you here.